test/libyul/yulOptimizerTests/fullSuite/reserved_identifiers.yul

```diff
@@ -16,14 +16,160 @@
 //
 // {
 //     {
-//         let dataoffset_ := datasize_(7)
-//         sstore(dataoffset_, g(9))
+//         let linkersymbol_1 := 0
```

Comment:
This test case has become much worse.

Reply:
Can you detect recursion?

Reply:
We could get that info from `SideEffectsCollector`. But we shouldn't just skip recursive functions, I think. If we do, then our fibonacci optimization will not work any more.

Ideally, the way to go ahead is by backtracking.

Reply:
Maybe it's still an option to just disable it on recursive functions for now, even though it hurts ;-), and try to work towards getting it back in a second step?

Reply:
I would agree here. Recursive functions should be extremely rare.